[2025-10-31T23:04:04.354+0700] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-10-31T23:04:04.370+0700] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: vehicle_pipeline_new_dag.start_vehicle_producer_cam1 scheduled__2025-10-31T15:00:00+00:00 [queued]>
[2025-10-31T23:04:04.373+0700] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: vehicle_pipeline_new_dag.start_vehicle_producer_cam1 scheduled__2025-10-31T15:00:00+00:00 [queued]>
[2025-10-31T23:04:04.373+0700] {taskinstance.py:2303} INFO - Starting attempt 3 of 3
[2025-10-31T23:04:04.380+0700] {taskinstance.py:2327} INFO - Executing <Task(BashOperator): start_vehicle_producer_cam1> on 2025-10-31 15:00:00+00:00
[2025-10-31T23:04:04.386+0700] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=2066) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-10-31T23:04:04.387+0700] {standard_task_runner.py:63} INFO - Started process 2080 to run task
[2025-10-31T23:04:04.388+0700] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'vehicle_pipeline_new_dag', 'start_vehicle_producer_cam1', 'scheduled__2025-10-31T15:00:00+00:00', '--job-id', '424', '--raw', '--subdir', 'DAGS_FOLDER/vehicel_pipeline_new_dag.py', '--cfg-path', '/tmp/tmppfqqy567']
[2025-10-31T23:04:04.391+0700] {standard_task_runner.py:91} INFO - Job 424: Subtask start_vehicle_producer_cam1
[2025-10-31T23:04:04.434+0700] {task_command.py:426} INFO - Running <TaskInstance: vehicle_pipeline_new_dag.start_vehicle_producer_cam1 scheduled__2025-10-31T15:00:00+00:00 [running]> on host f7bb1ad77e89
[2025-10-31T23:04:04.495+0700] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='vehicle_pipeline_new_dag' AIRFLOW_CTX_TASK_ID='start_vehicle_producer_cam1' AIRFLOW_CTX_EXECUTION_DATE='2025-10-31T15:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-10-31T15:00:00+00:00'
[2025-10-31T23:04:04.497+0700] {taskinstance.py:430} INFO - ::endgroup::
[2025-10-31T23:04:04.497+0700] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-10-31T23:04:04.498+0700] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', 'bash -c "timeout 45m bash /opt/***/projects/vehicle_count/scripts/run_producer_cam1.sh"']
[2025-10-31T23:04:04.508+0700] {subprocess.py:86} INFO - Output:
[2025-10-31T23:04:04.519+0700] {subprocess.py:93} INFO - [Producer-CAM1] ðŸš— Starting Vehicle Kafka Producer (Camera 1)...
[2025-10-31T23:04:04.541+0700] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-10-31T23:04:04.541+0700] {subprocess.py:93} INFO -   File "/opt/***/projects/vehicle_count/scripts/vehicle_producer_single.py", line 13, in <module>
[2025-10-31T23:04:04.542+0700] {subprocess.py:93} INFO -     import cv2
[2025-10-31T23:04:04.542+0700] {subprocess.py:93} INFO - ModuleNotFoundError: No module named 'cv2'
[2025-10-31T23:04:04.545+0700] {subprocess.py:97} INFO - Command exited with return code 1
[2025-10-31T23:04:04.545+0700] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-10-31T23:04:04.556+0700] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 460, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/bash.py", line 243, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-10-31T23:04:04.559+0700] {taskinstance.py:1205} INFO - Marking task as FAILED. dag_id=vehicle_pipeline_new_dag, task_id=start_vehicle_producer_cam1, execution_date=20251031T150000, start_date=20251031T160404, end_date=20251031T160404
[2025-10-31T23:04:04.566+0700] {standard_task_runner.py:110} ERROR - Failed to execute job 424 for task start_vehicle_producer_cam1 (Bash command failed. The command returned a non-zero exit code 1.; 2080)
[2025-10-31T23:04:04.601+0700] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-10-31T23:04:04.628+0700] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-10-31T23:04:04.629+0700] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-10-31T23:24:33.294+0700] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-10-31T23:24:33.308+0700] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: vehicle_pipeline_new_dag.start_vehicle_producer_cam1 scheduled__2025-10-31T15:00:00+00:00 [queued]>
[2025-10-31T23:24:33.311+0700] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: vehicle_pipeline_new_dag.start_vehicle_producer_cam1 scheduled__2025-10-31T15:00:00+00:00 [queued]>
[2025-10-31T23:24:33.311+0700] {taskinstance.py:2303} INFO - Starting attempt 3 of 3
[2025-10-31T23:24:33.318+0700] {taskinstance.py:2327} INFO - Executing <Task(BashOperator): start_vehicle_producer_cam1> on 2025-10-31 15:00:00+00:00
[2025-10-31T23:24:33.323+0700] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=3545) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-10-31T23:24:33.325+0700] {standard_task_runner.py:63} INFO - Started process 3551 to run task
[2025-10-31T23:24:33.326+0700] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'vehicle_pipeline_new_dag', 'start_vehicle_producer_cam1', 'scheduled__2025-10-31T15:00:00+00:00', '--job-id', '471', '--raw', '--subdir', 'DAGS_FOLDER/vehicel_pipeline_new_dag.py', '--cfg-path', '/tmp/tmprcau00z9']
[2025-10-31T23:24:33.328+0700] {standard_task_runner.py:91} INFO - Job 471: Subtask start_vehicle_producer_cam1
[2025-10-31T23:24:33.368+0700] {task_command.py:426} INFO - Running <TaskInstance: vehicle_pipeline_new_dag.start_vehicle_producer_cam1 scheduled__2025-10-31T15:00:00+00:00 [running]> on host f7bb1ad77e89
[2025-10-31T23:24:33.425+0700] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='vehicle_pipeline_new_dag' AIRFLOW_CTX_TASK_ID='start_vehicle_producer_cam1' AIRFLOW_CTX_EXECUTION_DATE='2025-10-31T15:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-10-31T15:00:00+00:00'
[2025-10-31T23:24:33.426+0700] {taskinstance.py:430} INFO - ::endgroup::
[2025-10-31T23:24:33.427+0700] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-10-31T23:24:33.427+0700] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', 'bash -c "timeout 45m bash /opt/***/projects/vehicle_count/scripts/run_producer_cam1.sh"']
[2025-10-31T23:24:33.440+0700] {subprocess.py:86} INFO - Output:
[2025-10-31T23:24:33.449+0700] {subprocess.py:93} INFO - [Producer-CAM1] ðŸš— Starting Vehicle Kafka Producer (Camera 1)...
[2025-10-31T23:24:33.471+0700] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-10-31T23:24:33.472+0700] {subprocess.py:93} INFO -   File "/opt/***/projects/vehicle_count/scripts/vehicle_producer_single.py", line 13, in <module>
[2025-10-31T23:24:33.472+0700] {subprocess.py:93} INFO -     import cv2
[2025-10-31T23:24:33.472+0700] {subprocess.py:93} INFO - ModuleNotFoundError: No module named 'cv2'
[2025-10-31T23:24:33.475+0700] {subprocess.py:97} INFO - Command exited with return code 1
[2025-10-31T23:24:33.475+0700] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-10-31T23:24:33.484+0700] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 460, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/bash.py", line 243, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-10-31T23:24:33.487+0700] {taskinstance.py:1205} INFO - Marking task as FAILED. dag_id=vehicle_pipeline_new_dag, task_id=start_vehicle_producer_cam1, execution_date=20251031T150000, start_date=20251031T162433, end_date=20251031T162433
[2025-10-31T23:24:33.493+0700] {standard_task_runner.py:110} ERROR - Failed to execute job 471 for task start_vehicle_producer_cam1 (Bash command failed. The command returned a non-zero exit code 1.; 3551)
[2025-10-31T23:24:33.538+0700] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-10-31T23:24:33.553+0700] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-10-31T23:24:33.554+0700] {local_task_job_runner.py:222} INFO - ::endgroup::
